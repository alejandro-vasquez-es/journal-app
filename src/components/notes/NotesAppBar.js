import React from 'react'

export const NotesAppBar = () => {
	return (
		<div className="notes__appbar">
			<span> 16 de febrero del 2022 </span>

			<div className='notes__btn-container'>
				<button className='btn'>
					Picture
				</button>
				<button className='btn'>
					Save
				</button>
			</div>
		</div>
	)
}
