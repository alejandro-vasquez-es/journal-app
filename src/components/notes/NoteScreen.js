import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
	return (
		<div className="notes__main-content">

			<NotesAppBar />

			<div className='notes__content'>

				<input
					type="text"
					placeholder="jijijajaja"
					className="notes__title-input"
				/>

				<textarea
					placeholder='what happened today'
					className='notes__textarea'
				>

				</textarea>

				<div className='notes__image'>
					<img
						alt='landcape'
						src='https://images.pexels.com/photos/592077/pexels-photo-592077.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
					/>
				</div>

			</div>

		</div>
	)
}
