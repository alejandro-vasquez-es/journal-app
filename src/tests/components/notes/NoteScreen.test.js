import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
// import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { NoteScreen } from '../../../components/notes/NoteScreen';
import { activeNote } from '../../../actions/notes';

jest.mock('../../../actions/notes', () => ({
	activeNote: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
	auth: {
		uid: 'ABCDEF',
		name: 'TESTER'
	},
	ui: {
		loading: false,
		msgError: null
	},
	notes: {
		notes: [{
			id: '123',
			date: 1646876753881,
			title: "TITLE",
			url: "https://res.cloudinary.com/dj54vlrsd/image/upload/v1646877128/su5hfvcu8mkmwn48ca6w.jpg",
			body: 'BODY'
		}],
		active: {
			id: '123',
			date: 1646876753881,
			title: "TITLE",
			url: "https://res.cloudinary.com/dj54vlrsd/image/upload/v1646877128/su5hfvcu8mkmwn48ca6w.jpg",
			body: 'BODY'
		},
	}
};
let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<NoteScreen />
	</Provider>
);


describe('Tests in <NoteScreen />', () => {

	test('should display propertly', () => {

		expect(wrapper).toMatchSnapshot();

	})

	test('should dispatch the activeNote', () => {

		wrapper.find('input[name="title"]').simulate('change', {
			targe: {
				name: 'title',
				value: 'TESTER TITLE'
			}
		});

		expect(activeNote).toHaveBeenLastCalledWith(
			'123', {
			date: 1646876753881,
			id: '123',
			title: "TITLE",
			url: "https://res.cloudinary.com/dj54vlrsd/image/upload/v1646877128/su5hfvcu8mkmwn48ca6w.jpg",
			body: 'BODY'
		}
		);

	})

})