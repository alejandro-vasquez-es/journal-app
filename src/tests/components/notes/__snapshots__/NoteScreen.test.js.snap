// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <NoteScreen /> should display propertly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
       16 de febrero del 2022 
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div
      className="notes__btn-container"
    >
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value="TITLE"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="what happened today"
      value="BODY"
    />
    <div
      className="notes__container"
    >
      <div
        className="notes__image"
      >
        <img
          alt="landcape"
          src="https://res.cloudinary.com/dj54vlrsd/image/upload/v1646877128/su5hfvcu8mkmwn48ca6w.jpg"
        />
      </div>
      <i
        className="fa-solid fa-trash-can notes__delete"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
